gtsfm/frontend/detector_descriptor/aslfeat.py

```diff
@@ -0,0 +1,95 @@
+"""Implements the ASLFeat detector-descriptor by wrapping around the original Pytorch source code.
+
+Paper: https://arxiv.org/abs/1905.03561
+https://github.com/mihaidusmanu/d2-net
+
+Authors: Travis Driver
+"""
+
+from pathlib import Path
+from typing import Tuple, Dict, Any
+import sys
+
+import numpy as np
+import tensorflow.compat.v1 as tf
+
+import gtsfm.utils.images as image_utils
+from gtsfm.common.image import Image
+from gtsfm.common.keypoints import Keypoints
+from gtsfm.frontend.detector_descriptor.detector_descriptor_base import DetectorDescriptorBase
+
+sys.path.append("thirdparty/ASLFeat")
+from thirdparty.ASLFeat.models.feat_model import FeatModel
+
+
+DEFAULT_MODEL_PATH = str(
+    Path(__file__).resolve().parent.parent.parent.parent
+    / "thirdparty"
+    / "ASLFeat"
+    / "weights"
+    / "aslfeat-cvgbedtrpjmou2-cleaned-open5"
+    / "model.ckpt-390000"
+)
+DEFAULT_CONFIG = {
+    "net": {
+        "max_dim": 2048,
+        "config": {
+            "kpt_n": 8000,
+            "kpt_refinement": True,
+            "deform_desc": 1,
+            "score_thld": 0.5,
+            "edge_thld": 10,
+            "multi_scale": True,
+            "multi_level": True,
+            "nms_size": 3,
+            "eof_mask": 5,
+            "need_norm": True,
+            "use_peakiness": True,
+        },
+    }
+}
+
+
+class ASLFeatDetectorDescriptor(DetectorDescriptorBase):
```

Comment:
shouldnt we add some unit tests for this?

Reply:
I was not able to get Tensorflow running on the CI